### PR TITLE
Add an ASE calculator

### PR DESCRIPTION
**Pull request by PabloPiaggi**
This pull request adds an ASE calculator for DeepMD-JAX. There is also an example of how to use it in the README file. I have tested that energies, forces, and virial are computed in agreement with the direct calculations using JAX-MD.

The calculator is useful to expand the use of DeePMD-JAX models beyond JAX-MD, which has a smaller user base (I think) compared to ASE.

I have only implemented the case without neighbor lists, but the performance is still quite reasonable (4 ns/day for 300 atoms).

One thing that may need attention before merging is that each call to the calculator triggers the execution of compute_lattice_candidates and that results in a message:
 `# Lattice vectors for neighbor images: Max %d out of %d candidates.`
Can we avoid that?

Otherwise, this is ready to be merged, in my opinion. Thanks.

### CODE REVIEW

**Comment by PabloPiaggi** [comment]
Hi Ruiqi, thanks for your answer and work, and sorry for the slow reply. I am trying to find time to test your modifications. I think I will be able to do this before this weekend. I will get back to you asap. Thanks again!

**Comment by PabloPiaggi** [comment]
Hi Ruiqi, I merged your commit to my branch. I carried out the following tests on an orthogonal box of water:
- I compared the energy computed using your code (i.e. `from deepmd_jax.train import test`) and the ASE calculator. **No problems**
- I compared the forces computed using your code and the ASE calculator. **No problems**
- I compared the analytical vs numerical stress tensor using the code I append below. **There were problems here**. The behavior changed with respect to my previous version of the code. The off-diagonal part of the numerical stress tensor is not computed properly anymore. The analytical stress tensor is still computed properly. The numerical stress tensor is based on changes in energy upon deformation, so I think this means that there are problems with the calculation of the energy when the box changes. In particular, the problem in the calculation of the energy happens when the box goes from orthogonal to non orthogonal, because diagonal components are OK. I am guessing the static_args are not handled properly in your new version.

Let me know what you think and if you understand how to fix this. The code should work well now for orthogonal boxes but not for non-orthogonal ones. Thanks!

Code for calculating the analytical and numerical stress tensor:
```python
import numpy as np
from ase import Atoms
from deepmd_jax.md import DPJaxCalculator
import jax
import jax.numpy as jnp

jax.config.update("jax_enable_x64", True)

# ==== USER INPUT ====
model_path = "model.pkl"
data_path = "./"
eps = 1e-4  # small strain magnitude

# ==== LOAD DATA ====
initial_position = np.genfromtxt(data_path + "/coord.raw")[0, :].reshape(-1, 3)
box = np.genfromtxt(data_path + "/box.raw")[0, :].reshape(3, 3)
type_idx = np.genfromtxt(data_path + "/type.raw").astype(int)
type_map = {0: "O", 1: "H"}
symbols = [type_map[i] for i in type_idx]

# ==== CREATE ATOMS & CALCULATOR ====
calc_dp = DPJaxCalculator(model_path=model_path, type_idx=type_idx, dtype=jnp.float64)
atoms = Atoms(symbols=symbols, positions=initial_position, cell=box, pbc=True)
atoms.calc = calc_dp

# ==== REFERENCE ENERGY ====
E0 = atoms.get_potential_energy()
V0 = atoms.get_volume()

print(f"Reference energy: {E0:.6f} eV, volume: {V0:.6f} Å³")

# ==== ANALYTICAL STRESS ====
stress_analytical = atoms.get_stress(voigt=False)  # ASE returns 3x3 if voigt=False
stress_analytical_GPa = stress_analytical * 160.21766208

print("\nAnalytical Stress (GPa):")
print(stress_analytical_GPa)

# ==== FINITE DIFFERENCE STRESS ====
# σ_ij = (1/V0) * ∂E / ∂ε_ij
# Compute by symmetric finite differences: (E(ε) - E(-ε)) / (2ε)

stress_fd = np.zeros((3, 3))

for i in range(3):
    for j in range(3):
        strain = np.zeros((3, 3))
        strain[i, j] = eps
 
        # +ε deformation
        box_plus = box @ (np.eye(3) + strain)
        atoms.set_cell(box_plus, scale_atoms=True)
        E_plus = atoms.get_potential_energy()
 
        # -ε deformation
        box_minus = box @ (np.eye(3) - strain)
        atoms.set_cell(box_minus, scale_atoms=True)
        E_minus = atoms.get_potential_energy()
 
        # finite difference derivative
        dE = (E_plus - E_minus) / (2 * eps)
        stress_fd[i, j] = dE / V0
 
        print(f"strain[{i},{j}]: dE = {dE:.6f} eV -> σ = {stress_fd[i,j]:.6e} eV/Å³")

# Symmetrize
#stress_fd = 0.5 * (stress_fd + stress_fd.T)

# ==== CONVERT TO GPa ====
# 1 eV/Å³ = 160.21766208 GPa
stress_fd_GPa = stress_fd * 160.21766208

print("\nFinite Difference Stress (GPa):")
print(stress_fd_GPa)

print("\nDifference (GPa):")
print(stress_fd_GPa - stress_analytical_GPa)
```

**Comment by SparkyTruck** [comment]
Hi Pablo,

Thanks for this interesting test case. I've update the code in the same [branch](https://github.com/SparkyTruck/deepmd-jax/commits/pr-21). It turns out that even in the previous code the analytical and numerical tensors matched, they were both wrong. The new numerical tensor, interestingly, is actually the correct result, and I've updated so that the analytical tensor matches it as well. It has something to do with the "ortho" variable inside static_args, which stand for orthonormal boxes. To get the stress tensor correct, even if the box is orthonormal, it cannot be set to a "ortho"=true workflow since the gradient workflow implies non-orthonormal directions. Let me know if the current code works properly!

Best,
Ruiqi
